### PR TITLE
Update the Migrate and Rollback commands with the new date option.

### LINKED ISSUE #117: Missing date option from Phinx v. 0.4.5
Since an update today to Phinx v. 0.4.5, Phinx has added the `date` option to migrations.

Phinx\Console\Command\Migrate::configure():
`->addOption('--date', '-d', InputOption::VALUE_REQUIRED, 'The date to migrate to')`

The update makes migrations throw the `InvalidArgumentException` since the `date` option is not defined in the CakePHP wrapper.

With the line below added to the CakePHP wrapper, everythings working again.

Migrations\Command\Migrate::configure():
`->addOption('--date', '-d', InputArgument::OPTIONAL, 'The date to migrate to')`

### PR DESCRIPTION
**Pull request by HavokInspiration**
The date option was added in Phinx 0.4.5 release and allow migrations to be migrated or rollbacked by specifying a date rather than targeting a specific migration

I also updated the classes to use the ``EventDispatcherTrait`` rather than the deprecated ``EventManagerTrait``.

I also fixed some docblocks.

Closes #117

### CODE REVIEW

**Comment by HavokInspiration** [comment]
Tests will probably fail, I think there is a logic issue with the date option. But I might not completely understand how the option is supposed to work.

I'll run some tests and open an issue on phinx tracker tomorrow.

File: tests/TestCase/MigrationsTest.php [hunk 1]
```diff
@@ -249,6 +249,83 @@ public function testOverrideOptions()
...
+                'status' => 'down',
+                'id' => '20150724233100',
+                'name' => 'UpdateNumbersTable'
+            ]
+        ];
+        $this->assertEquals($expectedStatus, $this->migrations->status());
+        $this->migrations->rollback([        ]);
+
+        $this->migrations->migrate([
+            'date' => '20150730',        ]);
```
Comment (markstory) [inline tests/TestCase/MigrationsTest.php]: Whitespace here is a bit funny.

File: tests/TestCase/MigrationsTest.php [hunk 2]
```diff
@@ -249,6 +249,83 @@ public function testOverrideOptions()
...
+                'id' => '20150704160200',
+                'name' => 'CreateNumbersTable'
+            ],
+            [
+                'status' => 'down',
+                'id' => '20150724233100',
+                'name' => 'UpdateNumbersTable'
+            ]
+        ];
+        $this->assertEquals($expectedStatus, $this->migrations->status());
```
Comment (markstory) [inline tests/TestCase/MigrationsTest.php]: Why is the status always the same after each operation?

File: src/Command/Migrate.php [hunk 3]
```diff
@@ -11,7 +11,7 @@
  */
 namespace Migrations\Command;
 
-use Cake\Event\EventManagerTrait;
+use Cake\Event\EventDispatcherTrait;
```
Comment (antograssiot) [inline src/Command/Migrate.php]: I guess that this also means that you need to bump the minimal required version of cakephp to 3.0.10 ?

File: tests/TestCase/MigrationsTest.php (hunk 2 above)
Comment (HavokInspiration) [inline tests/TestCase/MigrationsTest.php]: What do you mean ?

File: tests/TestCase/MigrationsTest.php (hunk 1 above)
Comment (HavokInspiration) [inline tests/TestCase/MigrationsTest.php]: Yeah, even for the rollback call above. Don't know what happend. I'll fix it.

File: src/Command/Migrate.php (hunk 3 above)
Comment (HavokInspiration) [inline src/Command/Migrate.php]: Derp, I did not thought about that. I'll revert it. Maybe when 3.1 is released we can bump the requirement to CakePHP 3.1 ?

Comment (lorenzo) [inline src/Command/Migrate.php]: That sounds like a good plan

**Comment by dereuromark** [comment]
We should probably ship this ASAP, as currently migrations cannot be used without locking the dependency to an older version.

**Comment by HavokInspiration** [comment]
It's my top priority. I'm currently having my lunch over testing. But I think I'll add a fix to the new methods in our Manager class and test it deeply afterwards to see if there really is a problem with the phinx implementation or if the problem is due to the fact that we are wrapping it and overriding options / arguments / methods / whatever.

**Comment by HavokInspiration** [comment]
It's my top priority. I'm currently having my lunch over testing. But I think I'll add a fix to the new methods in our Manager class and test it deeply afterwards to see if there really is a problem with the phinx implementation or if the problem is due to the fact that we are wrapping it and overriding options / arguments / methods / whatever.

It's a bit quick and dirty but will prevent the bug from happening. We can still remove the subclass methods afterwards.

**Comment by HavokInspiration** [comment]
I pushed some changes. Tests are passing locally with SQLite, I did not test with other db servers.

The problems I had with the implementation were :
- if you were to ask for a rollback to a date prior to the first migrations date, it did not rollback everything which seemed illogical to me. If I have two migrations : 2015 - 08 - 31 and 2015 - 08 - 28 and want to rollback to the state of 2015 - 08 - 25, I should get back to a blank state.
- if you were to ask for a migrate to a date prior the first migrations date, nothing should be migrated.

The subclass methods ``{migrate/rollback}ToDateTime`` implements this behavior.
If you think this behavior should not happened, I can delete the subclass methods and delete the associated tests.

**Comment by rafwell** [comment]
Same problem here

**Comment by eaxexe** [comment]
I am experiencing this issue as well, a merge/fix would be greatly appreciated!

**Comment by lorenzo** [comment]
Great job @HavokInspiration !

**Comment by rafwell** [comment]
Fixed! Thanx @HavokInspiration

**Pull request merged**